enterprise/backend/test/metabase_enterprise/task/cache_test.clj

```diff
@@ -468,97 +475,163 @@
             (is (= [] (t2/select :model/Query (@#'task.cache/duration-queries-to-rerun-honeysql
                                                [question-cache-config-1] true))))))))))
 
+(defn- compare-query-results
+  "Compares a normal query result with a query result generated after a preemptive caching job runs, and asserts
+  that all relevant fields are the same."
+  [original-result cached-result]
+  (let [clean-result (fn [result]
+                       (-> result
+                           (dissoc :running_time :average_execution_time :started_at :cached)
+                           (m/dissoc-in [:json_query :cache-strategy])))]
+    (is (= (clean-result original-result)
+           (clean-result cached-result)))))
```

Comment:
This is what catches the missing columns issue, right?

Reply:
Yeah, this + the e2e test